### ISSUE TITLE
💅 `noFloatingPromise` does not catch missing promises when called through Playwright fixtures

### ISSUE THREAD

**Issue by glassmancody-lcm** [biomejs/biome#11692]
### Environment information

<details>

```bash

CLI:
  Version:                      2.4.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.14.1
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.11.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    complexity/noAdjacentSpacesInRegex
    complexity/noExtraBooleanCast
    complexity/noUselessCatch
    complexity/noUselessEscapeInRegex
    complexity/noUselessTypeConstraint
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noSwitchDeclarations
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/useIsNan
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noFloatingPromises
    style/noEnum
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useConsistentMemberAccessibility
    style/useFilenamingConvention
    style/useNamingConvention
    style/useNumericSeparators
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCompareNegZero
    suspicious/noConstantBinaryExpressions
    suspicious/noDebugger
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noExplicitAny
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noImportAssign
    suspicious/noIrregularWhitespace
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noPrototypeBuiltins
    suspicious/noRedeclare
    suspicious/noShadowRestrictedNames
    suspicious/noSparseArray
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessRegexBackrefs
    suspicious/noWith
    suspicious/useGetterReturn
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0

```

</details>

### Rule name

noFloatingPromises

### Playground link

[Playground](https://biomejs.dev/playground/?lintRules=noFloatingPromises&tab=formatter&pane=Diagnostics#code=aQBtAHAAbwByAHQAIAB7ACAAdABlAHMAdAAgAGEAcwAgAGIAYQBzAGUAIAB9ACAAZgByAG8AbQAgACIAQABwAGwAYQB5AHcAcgBpAGcAaAB0AC8AdABlAHMAdAAiADsACgAKAGMAbwBuAHMAdAAgAHQAZQBzAHQAIAA9ACAAYgBhAHMAZQAuAGUAeAB0AGUAbgBkADwAewAgAGYAaQB4AHQAdQByAGUAOgAgACgAKQAgAD0APgAgAFAAcgBvAG0AaQBzAGUAPAB2AG8AaQBkAD4AIAB9AD4AKAB7AAoAIAAgAGYAaQB4AHQAdQByAGUAOgAgAGEAcwB5AG4AYwAgACgAewAgAH0ALAAgAHUAcwBlACkAIAA9AD4AIAB7AAoAIAAgACAAIABhAHcAYQBpAHQAIAB1AHMAZQAoAGEAcwB5AG4AYwAgACgAKQAgAD0APgAgAHsAIAB9ACkAOwAKACAAIAB9ACwACgB9ACkAOwAKAAoAdABlAHMAdAAoAGAAVABlAHMAdABgACwAIABhAHMAeQBuAGMAIAAoAHsAIABmAGkAeAB0AHUAcgBlACAAfQApACAAPQA%2BACAAewAKACAAIABmAGkAeAB0AHUAcgBlACgAKQA7AAoAfQApADsACgA%3D
)
### Expected result

> [!IMPORTANT]
> This example requires `npm i @playwright/test`
 
```typescript
import { test as base } from "@playwright/test";

class AsyncWrapper {
  public async doAsync() { }
}

const test = base.extend<{ example1: () => Promise<void>, example2: AsyncWrapper }>({
  example1: async ({ }, use) => {
    await use(async () => { });
  },
  example2: async ({ }, use) => {
    await use(new AsyncWrapper());
  }
});

test(`Test`, async ({ example1, example2 }) => {
  // Expected a noFloatingPromise error here
  example1();
  // Also expecting a noFloatingPromise error ehre
  example2.doAsync();
});
```

I expect a linting error to be thrown for the `example1()` and `example2.doAsync()` call, but none is.

I have tried to reproduce this without an external library (Playwright), but that failed as I misdiagnosed the issue and that led to a tangential bug being fixed in https://github.com/biomejs/biome/issues/11605

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
My clanker seems to think this is the minimum repro:

```ts
declare const x: { f<T>(): T };
x.f<Promise<void>>();
```

**Comment by dyc3** [comment]
Try this preview build when it finishes building: https://github.com/biomejs/biome/actions/runs/34632538322

**Comment by glassmancody-lcm** [comment]
> Try this preview build when it finishes building: https://github.com/biomejs/biome/actions/runs/34632538322

I downloaded the `cli-aarch64-apple-darwin.zip` artifact and ran it on the inked test case, it still fails to catch the floating promises.

It does fix the example you yourself linked, so must be another bug.

**Comment by dyc3** [comment]
Could you create a new issue with a new reproduction then? Since the reproduction here is fixed. Sorry about all the back and forth.

**Comment by glassmancody-lcm** [comment]
> Could you create a new issue with a new reproduction then? Since the reproduction here is fixed. Sorry about all the back and forth.

what would be new issue exactly? I can still reproduce with the original test case.

**Comment by ematipico** [comment]
@dyc3 I wouldn't chase what the clanker says. OP submitted a bug for that particular code and that's what we should try to fix

**Comment by dyc3** [comment]
Oh, I missunderstood what you said @glassmancody-lcm. I was able to see the original repro fixed on my end, but maybe I made a mistake

**Comment by dyc3** [comment]
I just tried it again. I'm no longer able to reproduce the bug with the fix in #11737 applied and using `@playwright/test@1.63.0`

<img width="1286" height="955" alt="Image" src="https://github.com/user-attachments/assets/f3933e12-7db5-45a1-9974-9b2178c649bf" />

@glassmancody-lcm can you confirm if you ran the preview build against your project or against the posted original reproduction? What version of `@playwright/test` are you using? If you could create a repo that reproduces the bug (using the preview build), that would eliminate the ambiguity here.

**Comment by glassmancody-lcm** [comment]
@dyc3 Ah yeah after pinning version and re-trying with the build I do see both issues fixed. Thanks, it also seems to catch some other floating promises in our project which is great news!
